File: w10_d1_0_white_boarding/solutions/3_calc_change.rb
# Problem 3 - Calculate Change

def calc_change(cost, money)
  cost *= 100
  money *= 100

  change_types = {
    '$20s' => 2000,
    '10s' => 1000,
    '$5s' => 500,
    '$1s' => 100,
    'quarters' => 25,
    'dimes' => 10,
    'nickels' => 5,
    'pennies' => 1,
  }
  total_change = money - cost
  change_counts = {}

  # iterate through change_types
  change_types.each do |name, value|
    # calculate count of change_type needed
    count = (total_change / value).floor
    # save count in change_counts hash with name as key
    change_counts[name] = count
    # subtract amount of current change_type from total_change
    total_change -= count * value
  end

  change_counts
end

File: w10_d1_0_white_boarding/solutions/3_calc_change.js
// Problem 3 - Calculate Change

function calcChange(cost, money) {
  cost *= 100;
  money *= 100;

  var changeNames = ['$20s', '$10s', '$5s', '$1s', 'quarters', 'dimes', 'nickels', 'pennies'];
  var changeTypes = [2000, 1000, 500, 100, 25, 10, 5, 1];
  var totalChange = money - cost;
  var changeCounts = {};

  // loop through changeTypes
  for(var i = 0; i < changeTypes.length; i++) {
    // calculate count of changeType needed
    var count = Math.floor(totalChange / changeTypes[i]);
    // save count in changeCounts object with changeName as key
    changeCounts[changeNames[i]] = count;
    // subtract amount of current changeType from totalChange
    totalChange -= count * changeTypes[i];
  }
  return changeCounts;
}